### PR TITLE
feat: refactor routes to handle 404 errors

### PR DESCRIPTION
**Pull request by j0hnfl0w**
- like in [snapshot](https://github.com/snapshot-labs/snapshot/blob/develop/src/router.ts) we need to handle 404 error
- we can validate our composite space ID = [networkId, spaceId] with routerGuard validateIdGuard
- we can group all space related routes to space route children
- minor fixes in Proposal view and Proposal component

### CODE REVIEW

File: src/router.ts [hunk 1]
```diff
@@ -10,8 +10,25 @@ import Proposal from '@/views/Proposal.vue';
...
+import { isAddress } from '@ethersproject/address';
+import { getNetwork } from '@/networks';
+
+const validateIdGuard = (to, from, next) => {
+  try {
+    const id = to.params.id;
+    const [networkId, spaceId] = id.split(':');
+    console.log(':beforeEnter,', networkId, spaceId);
+    getNetwork(networkId);
+    if (!isAddress(spaceId)) return next('/');
```
Comment (Sekhmet) [inline src/router.ts:22]: This won't work for starknet spaces.

File: src/router.ts [hunk 2]
```diff
@@ -10,8 +10,25 @@ import Proposal from '@/views/Proposal.vue';
...
 import Create from '@/views/Create.vue';
 
+import { isAddress } from '@ethersproject/address';
+import { getNetwork } from '@/networks';
+
+const validateIdGuard = (to, from, next) => {
+  try {
+    const id = to.params.id;
+    const [networkId, spaceId] = id.split(':');
+    console.log(':beforeEnter,', networkId, spaceId);
```
Comment (Sekhmet) [inline src/router.ts:20]:
```suggestion
```

**Review by Sekhmet — CHANGES_REQUESTED** [review]

File: src/router.ts (hunk 1 above)
Comment (j0hnfl0w) [reply src/router.ts:22]: So we don't need this validation then

**Comment by j0hnfl0w** [comment]
So now this PR adds only grouping Space routes in children and 404 redirect to /

File: src/router.ts (hunk 1 above)
Comment (Sekhmet) [reply src/router.ts]: Maybe we do, we could validate both networks if we wanted. I don't know what we want, it was your idea I think :D

Comment (j0hnfl0w) [reply src/router.ts]: I think we can add it later for both networks

File: src/router.ts [hunk 3]
```diff
@@ -21,18 +23,21 @@ const routes: any[] = [
...
+    path: '/profile/:id',
+    name: 'user',
+    component: User
   },
-  { path: '/:space/proposal/:id?', name: 'proposal', component: Proposal },
-  { path: '/profile/:id', name: 'user', component: User },
-  { path: '/create', name: 'create', component: Create },
-  { path: '/explore', name: 'explore', component: Explore }
+  {
+    path: '/:pathMatch(.*)*',
```
Comment (Sekhmet) [inline src/router.ts:37]: Maybe we can whitelist `:id` in `space` route, so it doesn't math anything that doesn't look like space ID and swallow those.
```
path: '/:id([^:]+:[A-Za-z0-9]+)',
```

Comment (Sekhmet) [reply src/router.ts:37]: @j0hnfl0w what do you think?

**Review by Sekhmet — APPROVED** [review]
tACK

**Pull request merged**